internal/provider/resources/deployment_test.go

```diff
@@ -95,6 +103,7 @@ resource "prefect_deployment" "{{.DeploymentName}}" {
 	version = "{{.Version}}"
 	work_pool_name = "{{.WorkPoolName}}"
 	work_queue_name = "{{.WorkQueueName}}"
+	global_concurrency_limit_id = {{if .GlobalConcurrencyLimitID}}prefect_global_concurrency_limit.test_limit.id{{else}}null{{end}}
```

Comment:
```suggestion
  {if .GlobalConcurrencyLimitID}}
	global_concurrency_limit_id = prefect_global_concurrency_limit.test_limit.id
	{{end}}
```

What do you think of this approach? Might be a little more legible.

Reply:
To test unsetting the global_concurrency_limit_id you have to explicitly send null. I don't think this if statement does that.